source/agora/node/FullNode.d

```diff
@@ -867,4 +868,31 @@ public class FullNode : API
         }
         return found_txs;
     }
+
+    /***************************************************************************
+
+        Params:
+            heights = Set of block Heights to return header for
+
+        Returns:
+            BlockHeader if the client has that block
+
+    ***************************************************************************/
+
+    public BlockHeader[] getBlockHeaders (Set!ulong heights) @safe
+    {
+        import std.algorithm: min;
+        import std.conv;
+
+        BlockHeader[] headers;
+        if (!heights.empty)
+        {
+            foreach (block; this.ledger.getBlocksFrom(Height(heights[].front)))
+            {
+                if (block.header.height in heights)
+                    headers ~= block.header;
+            }
+        }
+        return headers;
+    }
```

Comment:
This is an expensive API. We really need to improve it, but let's go with it for now.